internal/services/resource/resource_provider_registration_resource_test.go

```diff
@@ -123,6 +132,65 @@ resource "azurerm_resource_provider_registration" "import" {
 `, template)
 }
 
+func unRegisterProvider() error {
```

Comment:
Can we make this a method of `ResourceProviderRegistrationResource{}`

```suggestion
func (ResourceProviderRegistrationResource) unRegisterProvider() error {
```